docs/api-reference.md

```diff
@@ -514,17 +690,102 @@ for _, row := range result.Values {
 }
 ```
 
+### Performance Optimization
+
+#### Cache Strategy
+
+The SDK provides intelligent caching to optimize query performance:
+
+**When to Use Caching:**
+- Repeated queries with identical parameters
+- Dashboard or monitoring applications
+- Data visualization with frequent refreshes
+- Batch processing where data consistency is acceptable
+
+**Cache Behavior:**
+- Cache keys include all query parameters (DataProvider, StreamId, From, To, FrozenAt, BaseDate)
+- Cache TTL is configurable server-side
+- Cache invalidation occurs when new data is inserted
+- Cache statistics are returned with every query
```

Comment:
I guess this part might be a little off, as our cache is a bit different from parameter caching

i.e. our cache is about pre-configured streams by the operators, with no invalidation on new data, but the cache refreshes periodically (depending on each node operators, so they should reach to them if they desire more cached streams or even host their own node)

The only dependency that it has on parameters is that when frozenat or basedate parameter is specified, the cache is bypassed, as we don't support those

Maybe TTL isn't the best term for this refresh behavior (i.e. it doesn't expire, but we do return the cached date for the user to determine himself the acceptable freshness)